src/components/OSCALLoader.js

```diff
@@ -120,37 +85,16 @@ export default function OSCALLoader(props) {
       );
   };
 
-  /**
-   * Sends a REST request of type restMethod to a backend service and updates the viewer if
-   * the request is successful.
-   *
-   * @param appendToLastFieldInPath boolean indicating if the updated value should be appended to an array or replace an existing value
-   * @param partialRestData data that will be passed into the body of the REST request, may not initially contain the updates
-   * @param editedFieldJsonPath path to the field that is being updated
-   * @param newValue updated value for the edited field
-   * @param httpMethod the HTTP request type
-   * @param restUrlPath path defining where in the file the modifications are made
-   * @param jsonRootName root OSCAL object, as it appears on the corresponding object file, of the JSON file
-   * @param restPath main url path for access the OSCAL files in REST mode
-   */
-  const handleRestRequest = (
+  const handleFieldSave = (
```

Comment:
Something named `handleRestRequest` should really only do that, but here we had a single function manipulating the data, building the request, and performing the request, and defining behaviors for success and failure.

`handleFieldSave` is still responsible for all of that, but delegates isolated bits of that functionality to other functions.